### PR TITLE
Feature/page display for sources #67

### PR DESCRIPTION
**Pull request by shishi5089**

### CODE REVIEW

File: src/UI/components/sources/source_form.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        register, 
+        control,
+        handleSubmit, 
+        watch, 
+        formState:{errors}} = useForm<IFormInputs>({
+          resolver: yupResolver(schema)
+        });
+
+      console.log('errors', errors);
+      console.log('watch variable author', watch('author'));
```
Comment (Christopher-Sandbach) [inline src/UI/components/sources/source_form.tsx]: Probably best to remove the console logs

File: src/UI/components/sources/source_form.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+      const formSubmitHandler: SubmitHandler<IFormInputs> = (data: IFormInputs) => {
+        console.log('Form data is ', data)
```
Comment (Christopher-Sandbach) [inline src/UI/components/sources/source_form.tsx]: Here also

File: src/UI/components/sources/source_form.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+          control={control}
+          render= {({field}) => (
+          <TextField {...field} 
+          type='text'
+          label= "Author:" 
+          variant='outlined'
+          />
+          )}/>
+          <br />
+       {/* <input {...register('author')}/> */}
```
Comment (Christopher-Sandbach) [inline src/UI/components/sources/source_form.tsx]: Same with the jsx comments

File: src/UI/components/sources/source_form.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        <Controller 
+          name="journal_title" 
+          control={control}
+          render= {({field}) => (
+          <TextField {...field} 
+          type='text'
+          label= "Journal title:" 
+          variant='outlined'/>
+          )}/><br />
+        {/* <input {...register('journal_title')}/> */}
```
Comment (Christopher-Sandbach) [inline src/UI/components/sources/source_form.tsx]: jsx comment

File: src/UI/components/sources/source_form.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        control={control}
+        render= {({field}) => (
+          <TextField{...field} 
+          label= 'Citation:' 
+          variant='outlined'
+          error={!!errors.citation}
+          helperText={errors.citation ? errors.citation?.message : ''} />
+        )} />
+        </div>
+        <br /> */}
```
Comment (Christopher-Sandbach) [inline src/UI/components/sources/source_form.tsx]: jsx comment

File: src/UI/components/sources/source_form.tsx [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+        <Controller 
+          name="year" 
+          control={control}
+          render= {({field}) => (
+          <TextField {...field} 
+          type='number'
+          label= "Year:" 
+          variant='outlined'/>
+          )}/><br />
+        {/* <input {...register('year')}/> */}
```
Comment (Christopher-Sandbach) [inline src/UI/components/sources/source_form.tsx]: jsx comment here too

File: src/UI/components/sources/source_form.tsx [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        <Controller 
+          name="report_type" 
+          control={control}
+          render= {({field}) => (
+          <TextField {...field} 
+          type='text'
+          label= "Report type:" 
+          variant='outlined'/>
+          )}/><br />
+        {/* <input {...register('report_type')}/> */}
```
Comment (Christopher-Sandbach) [inline src/UI/components/sources/source_form.tsx]: and here

File: src/UI/components/sources/source_form.tsx [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+            <><FormLabel>Published: </FormLabel><RadioGroup
+              defaultValue=""
+            >
+              <FormControlLabel value="True" control={<Radio />} label="True"></FormControlLabel>
+              <FormControlLabel value="False" control={<Radio />} label="False"></FormControlLabel>
+            </RadioGroup></>
+          
+          )}/>
+          <br />
+        {/* <input {...register('published')}/> */}
```
Comment (Christopher-Sandbach) [inline src/UI/components/sources/source_form.tsx:182]: and here

File: src/UI/components/sources/source_form.tsx [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+          control={control}
+          render= {({field}) => (
+            <><FormLabel>Vector Data: </FormLabel><RadioGroup
+            defaultValue=""
+          >
+            <FormControlLabel value="True" control={<Radio />} label="True"></FormControlLabel>
+            <FormControlLabel value="False" control={<Radio />} label="False"></FormControlLabel>
+          </RadioGroup></>
+          )}/>
+        {/* <input {...register('v_data')}/> */}
```
Comment (Christopher-Sandbach) [inline src/UI/components/sources/source_form.tsx:207]: And then here as well. Thanks

File: src/UI/package.json [hunk 10]
```diff
@@ -13,28 +13,32 @@
...
     "@emotion/styled": "^11.10.0",
+    "@hookform/resolvers": "^2.9.8",
+    "@material-ui/core": "^4.12.4",
     "@mui/icons-material": "^5.8.4",
     "@mui/material": "^5.9.3",
     "@reduxjs/toolkit": "^1.8.3",
     "@typescript-eslint/eslint-plugin": "^5.30.7",
     "axios": "^0.27.2",
-    "graphql": "^16.5.0",
-    "js-file-download": "^0.4.12",
```
Comment (Christopher-Sandbach) [inline src/UI/package.json:27]: I think we want to be keeping this. At least for now as it's what we use to download data from the main page

File: src/UI/pages/source_page.tsx [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+import { useDispatch } from 'react-redux';
+import SingleSource from '../components/sources/singleSource';
+
+
+
+function SourcesView(): JSX.Element{
+    const source_info = useAppSelector((state) => state.source.source_info)
+    const source_info_status = useAppSelector((state) => state.source.source_info_status)
+
+    console.log(source_info);
```
Comment (Christopher-Sandbach) [inline src/UI/pages/source_page.tsx]: Console here

Comment (shishi5089) [reply src/UI/pages/source_page.tsx]: Thanks so much. Will do that

Comment (Christopher-Sandbach) [reply src/UI/pages/source_page.tsx]: No worries. If I can get my git to behave I'll have a look at it locally too - but I'm getting an invalid path which is fun

**Comment by andykitch** [comment]
Please can you remove all the package-lock backups?

**Comment by andykitch** [comment]
Please remove the `\wsl.localhostUbuntuhomeshishiVECTOR_ATLASvectoratlas-software-code` file

**Comment by andykitch** [comment]
Why has the src/API/.envrc.template file been removed?

**Comment by andykitch** [comment]
Why have the files in `src/TileServer` been changed?

File: src/UI/.env.local.template [hunk 12]
```diff
@@ -9,7 +9,7 @@ AUTH0_ISSUER_BASE_URL='https://dev-326tk4zu.us.auth0.com'
 AUTH0_CLIENT_ID='Al9JO5wSIRNMf8kv5zl2KBW1l5iT2coW'
 # Your Auth0 application's Client Secret
 # This is currently stored in NPD, under Technical Documents - Technical Report - Auth0 Secrets for Vector Atlas
-AUTH0_CLIENT_SECRET='<auth0-client-secret>'
+AUTH0_CLIENT_SECRET='xDAJO3Jh0PKvFeYjDEy01frdktCI5hELOqWWKPTmt_rzu2xWhh8pqvHYtjF2r7Yg'
```
Comment (andykitch) [inline src/UI/.env.local.template:12]: Please can you revert this change - secrets should not be checked into the codebase,

File: src/UI/.env.local.template.orig [hunk 13]
```diff
@@ -0,0 +1,16 @@
+# A long, secret value used to encrypt the session cookie
```
Comment (andykitch) [inline src/UI/.env.local.template.orig:1]: I don't think we need this file checked in?

File: src/UI/.env.production.template [hunk 14]
```diff
@@ -9,5 +9,5 @@ AUTH0_ISSUER_BASE_URL='https://dev-326tk4zu.us.auth0.com'
 AUTH0_CLIENT_ID='Al9JO5wSIRNMf8kv5zl2KBW1l5iT2coW'
 # Your Auth0 application's Client Secret
 # This is currently stored in NPD, under Technical Documents - Technical Report - Auth0 Secrets for Vector Atlas
-AUTH0_CLIENT_SECRET='<auth0-client-secret>'
+AUTH0_CLIENT_SECRET='xDAJO3Jh0PKvFeYjDEy01frdktCI5hELOqWWKPTmt_rzu2xWhh8pqvHYtjF2r7Yg'
```
Comment (andykitch) [inline src/UI/.env.production.template:12]: Please revert changes to this file too.

File: src/UI/.env.production.template.orig [hunk 15]
```diff
@@ -0,0 +1,16 @@
+# A long, secret value used to encrypt the session cookie.
```
Comment (andykitch) [inline src/UI/.env.production.template.orig:1]: I don't think we need this file checked in?

**Comment by andykitch** [comment]
Please can you add some unit tests to the changed and added code, both in the UI and API. The existing unit test files should have examples you can follow.

File: src/UI/components/sources/singleSource.tsx [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+          borderRadius: 10,
+          border: 2,
+          borderColor: 'primary.main',
+          boxSizing: 'inherit',
+          marginBottom: 5,
+        }}
+      >
+        <Grid item md={12} lg={4} xs={6}>
+          <Typography variant="h6" color="primary" py="15px" px="15px">
+            {' '}
```
Comment (andykitch) [inline src/UI/components/sources/singleSource.tsx:38]: Why are these empty strings here?

File: src/UI/components/sources/source_form.tsx [hunk 17]
```diff
@@ -0,0 +1,194 @@
...
+import { useForm, SubmitHandler, Controller } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+  Radio,
+  RadioGroup,
+  FormControl,
+  FormLabel,
+  FormControlLabel,
+  TextField,
+} from '@material-ui/core';
```
Comment (andykitch) [inline src/UI/components/sources/source_form.tsx:11]: We should be using the `@mui` package for these things - material-ui has been replaced my mui

File: src/UI/pages/new_source.tsx [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+
+function NewSource() {
+  const role = useAppSelector((state) => state.auth.roles);
+
+  if (role.includes('uploader')) {
+    return (
+      <>
+        <Paper
+          sx={{
+            // display: 'flex',
```
Comment (andykitch) [inline src/UI/pages/new_source.tsx:16]: Can we avoid commented-out code please?

File: src/UI/pages/source_page.tsx [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+import { Paper, Typography, Box, Button, Container, Grid } from '@mui/material';
+import { getSourceInfo } from '../state/sourceSlice';
+import { AppDispatch } from '../state/store';
+import { useAppSelector } from '../state/hooks';
+import { useDispatch } from 'react-redux';
+import SingleSource from '../components/sources/singleSource';
+
+function SourcesView(): JSX.Element {
+  const source_info = useAppSelector((state) => state.source.source_info);
+  const source_info_status = useAppSelector(
```
Comment (andykitch) [inline src/UI/pages/source_page.tsx:11]: Is `source_info_status` actually being used?

File: src/UI/state/sourceSlice.ts [hunk 20]
```diff
@@ -0,0 +1,55 @@
+import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
+import axios from 'axios';
```
Comment (andykitch) [inline src/UI/state/sourceSlice.ts:2]: I don't think `axios` is being used here

File: src/UI/state/sourceSlice.ts [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+  extraReducers: (builder) => {
+    builder
+    .addCase(getSourceInfo.pending, (state) => {
+      state.source_info_status = 'loading';
+    })
+    .addCase(getSourceInfo.rejected, (state, action) => {
+      state.source_info_status = 'error';
+    })
+    .addCase(getSourceInfo.fulfilled, (state, action) => {
+      state.source_info = action.payload;
```
Comment (andykitch) [inline src/UI/state/sourceSlice.ts:49]: If you do use `source_info_status` anywhere, setting it to something like `success` would be good here.

**Comment by andykitch** [comment]
Why have the `buildVersionFiles.sh` and `generateTokenKey.sh` files been changed?

**Comment by andykitch** [comment]
Please remove the package-lock.json` file from the src root directory

File: src/UI/pages/source_page.tsx [hunk 22]
```diff
@@ -0,0 +1,59 @@
+import { useEffect } from 'react';
```
Comment (andykitch) [inline src/UI/pages/source_page.tsx:1]: Can we rename this file from `source_page` to `source_list` please?

File: src/UI/pages/new_source.tsx [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+                  </Typography>
+                  <br />
+                  <SourceForm />
+                </div>
+              </Container>
+            </main>
+          </div>
+        </Paper>
+      </>
+    );
```
Comment (andykitch) [inline src/UI/pages/new_source.tsx:44]: Looking at this page in the browser, it looks like the paper component wraps the whole width of the page, not just the content. Can this be changed so it looks more like the about page?

File: src/UI/components/sources/source_form.tsx [hunk 24]
```diff
@@ -0,0 +1,194 @@
...
+            </>
+          )}
+        />
+      </div>
+      <br />
+
+      <Button variant="contained" size="large">
+        SUBMIT
+      </Button>
+    </form>
```
Comment (andykitch) [inline src/UI/components/sources/source_form.tsx:192]: When I try to submit a new source, I get a console error in the browser - `Warning: A component is changing an uncontrolled input to be controlled.` Do you see this too?

File: src/UI/pages/source_page.tsx [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+                </Typography>
+
+                {source_info.map((source_info) => (
+                  <SingleSource key={source_info.citation} {...source_info} />
+                ))}
+              </div>
+            </Grid>
+          </Container>
+        </Paper>
+      </main>
```
Comment (andykitch) [inline src/UI/pages/source_page.tsx:54]: The same comment applies to this as to the `new_source` page - the paper is wrapping the whole width

File: src/API/src/db/shared/reference.resolver.ts [hunk 26]
```diff
@@ -0,0 +1,18 @@
...
+  constructor(private referenceService: ReferenceService) {}
+
+  @Query(() => Reference)
+  async referenceData(@Args('id', { type: () => String }) id: string) {
+    return this.referenceService.findOneById(id);
+  }
+
+  @Query(() => [Reference])
+  async allReferenceData() {
+    return this.referenceService.findAll();
```
Comment (andykitch) [inline src/API/src/db/shared/reference.resolver.ts:16]: The acceptance criteria for this story (https://github.com/icipe-official/vectoratlas-software-code/issues/67) mention adding infinite scroll - is there a plan to add this in later?

**Comment by andykitch** [comment]
What stories does this PR cover?

File: src/UI/pages/new_source.tsx [hunk 27]
```diff
@@ -1,25 +1,24 @@
...
 import { useAppSelector } from '../state/hooks';
 
+
+
 function NewSource() {
   const role = useAppSelector((state) => state.auth.roles);
+ 
+  // function handleSubmit(event){
+  //   event
+  // }
```
Comment (andykitch) [inline src/UI/pages/new_source.tsx:14]: Please remove these lines if they're not needed!

File: src/UI/pages/source_list.tsx [hunk 28]
```diff
@@ -5,22 +5,41 @@ import { AppDispatch } from '../state/store';
...
+  const refresh = () => {};
+
   return (
     <>
+    <InfiniteScroll 
+        dataLength={source_info.length}
+        next={() => {
+          dispatch(getSourceInfo());
+        } }
+        hasMore={false}
```
Comment (andykitch) [inline src/UI/pages/source_list.tsx:27]: If `hasMore` is always set to false, it's never going to get more data... Is this intended?

File: src/API/.envrc.template [hunk 29]
```diff
@@ -6,4 +6,4 @@ export POSTGRES_HOST="127.0.0.1"
 export PORT=7000
 export AUTH0_ISSUER_URL='https://dev-326tk4zu.us.auth0.com/'
 export AUTH0_AUDIENCE='https://www.vectoratlas.org'
-export TOKEN_KEY='<token_key>'
\ No newline at end of file
+export TOKEN_KEY='BkpHYvWSj4OfCi1FhdAXw8VCBQ0='
```
Comment (andykitch) [inline src/API/.envrc.template:9]: Please can you undo this change?

File: src/UI/api/queries.ts [hunk 30]
```diff
@@ -1,3 +1,4 @@
+import { gql } from '@apollo/client';
```
Comment (andykitch) [inline src/UI/api/queries.ts:1]: I don't think `gql` is being used here

File: src/UI/package.json [hunk 31]
```diff
@@ -13,17 +13,20 @@
     "test:watch": "jest --watch"
   },
   "dependencies": {
+    "@apollo/client": "^3.7.0",
```
[the remark this answers is not in the document]
Comment (shishi5089) [reply src/UI/package.json:16]: Do I delete them directly?

**Review by andykitch — APPROVED** [review]
